sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/encoders/EncoderResolutionSuite.scala

```diff
@@ -196,6 +196,43 @@ class EncoderResolutionSuite extends PlanTest {
     encoder.resolveAndBind(attrs)
   }
 
+  test("SPARK-28497: complex type is not compatible with string encoder schema") {
+    val encoder = ExpressionEncoder[String]
+
+    {
+      val attrs = Seq('a.struct('x.long))
+      assert(intercept[AnalysisException](encoder.resolveAndBind(attrs)).message ==
+        s"""
```

Comment:
super nit: its ok just check `.contains("Cannot up cast `a` from struct<x:bigint> to string")`?

Reply:
+1

Reply:
I am OK with either way. I was following the other test cases in the suite. Let me change it to `.contains...`